File: projects/team-adventure/team/be/src/domain/PlayerRepository.js
// Domain Repository: In-Memory Player Repository
class PlayerRepository {
  constructor() {
    this.players = {};
  }

  save(player) {
    this.players[player.id] = player;
    return player;
  }

  findById(id) {
    return this.players[id] || null;
  }

  findAll() {
    return { ...this.players };
  }

  delete(id) {
    if (this.players[id]) {
      const player = this.players[id];
      delete this.players[id];
      return player;
    }
    return null;
  }
}

module.exports = PlayerRepository;

File: team-adventure-be/src/main/java/com/teamadventure/domain/repository/PlayerRepository.java
package com.teamadventure.domain.repository;

import com.teamadventure.domain.model.Player;
import java.util.Collection;
import java.util.Optional;

public interface PlayerRepository {
  Player save(Player player);

  Optional<Player> findById(String id);

  Collection<Player> findAll();

  Optional<Player> remove(String id);
}
